python/pyarrow/array.pxi

```diff
@@ -721,12 +722,12 @@ cdef class _PandasConvertible(_Weakrefable):
         integer_object_nulls : bool, default False
             Cast integers with nulls to objects
         date_as_object : bool, default True
-            Cast dates to objects. If False, convert to datetime64[ns] dtype.
+            Cast dates to objects. If False, convert to datetime64 dtype.
```

Comment:
Do we want to specify this is [ns] for older pandas and [ms] for pandas >= 2.0?

Reply:
I will add the case for pandas 1.x! For pandas 2.0, the conversion will match the pyarrow time unit so it could range from `s` to `ns`. The only surprise from the user's perspective is date32, which translates `Day` to `ms`.

Reply:
> For pandas 2.0, the conversion will match the pyarrow time unit so it could range from `s` to `ns`

This is for dates (so date32/date64), so isn't that always "ms" unit?